ripe/atlas/tools/commands/stream.py

```diff
@@ -50,12 +63,22 @@ def add_arguments(self):
                  "appropriate renderer will be selected."
         )
 
+    def _get_request_auth(self):
+        if self.arguments.auth:
+            return conf["authorisation"]["fetch_aliases"][self.arguments.auth]
+        else:
+            return conf["authorisation"]["fetch"]
+
     def run(self):
 
         try:
             measurement = Measurement(
-                id=self.arguments.measurement_id, user_agent=self.user_agent)
-        except APIResponseError:
+                id=self.arguments.measurement_id, user_agent=self.user_agent,
+                key=self._get_request_auth)
+        except APIResponseError as e:
+            if "error" in e.args[0]:
+                if "detail" in e.args[0]["error"]:
+                    raise RipeAtlasToolsException(e.args[0]["error"]["detail"])
             raise RipeAtlasToolsException("That measurement does not exist")
```

Comment:
Why not raise the whole e.args[0] instead of stripping it down. I think the whole message has it's value.